Fix itest HTTP remote peer creation panic

kylehuntsman opened the pull request:
Ran into a CI test error where there was a panic in [`generator.go`](https://github.com/filecoin-project/lassie/blob/92924796eba357c3412beb471f661f7c5d846c52/pkg/internal/itest/testpeer/generator.go#L273) when creating an HTTP remote peer when the address was already bound to. With the way the mocknet `Identity` struct is instantiated, the random port that is assigned to it's multiaddrress has the potential to be already bound.

From  [go-libp2p-testing/net/gen.go](https://github.com/libp2p/go-libp2p-testing/blob/aac9ce834b218b0d51ab729256812a1b4ea9164b/net/gen.go#L29-L33):
> // chances are it will work out, but it **might** fail if the port is in use
> // most ports above 10000 aren't in use by long running processes, so yay.
> // (maybe there should be a range of "loopback" ports that are guaranteed
> // to be open for the process, but naturally can only talk to self.)

hannahhoward commented:
what is ctx.Done() for? this just retrieves the channel.

Also the second error check doesn't appear to do anything -- error hasn't changed since way up above

rvagg commented:
oh yeah, good catch @hannahhoward, I probably should have picked those up in original review. Here's my suggestion list:

* Move everything but the `Start` out of the goroutine so you can return the first `err`
* Ditch the `ctx.Done()`s entirely
* `if err := peerHttpServer.Start(); err != http.ErrServerClosed {` for that second error case and use `t` to handle it since you have it on hand, `t.Fatal` is probably enough.

hannahhoward commented:
@kylehuntsman should we close this? Seems like it's non urgent and no one is getting to it.

kylehuntsman commented:
> * use `t` to handle it since you have it on hand, `t.Fatal` is probably enough.

Compiler was complaining that I can't use `t.Fatal` since it's not being called in the same goroutine as the test.

I realized that I was being dumb and the TestPeerHttpServer already had a start method handling the `ErrServerClosed`. Simplified the goroutine to just call `Start()`, as it handles logging and returning other errors. Added another routine for closing the server on context cancel.
